test-apps/appui-test-app/appui-test-providers/src/Translation.ts

```diff
@@ -0,0 +1,22 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AppUiTestProviders } from "./AppUiTestProviders";
+
+interface Localization {
+  translate: (key: string) => string;
+}
+
+/**
+ * Returns a localization object.
+ * @internal
+ */
+export function useTranslation(): Localization {
```

Comment:
Should we rename `Localization`? Or use an an implicit type (remove `Localization`, specify nothing as return type in `useTranslation() {}`

Either way, I think we need a consistent naming strategy for hooks (might be a good addition to CONTRIBUTING).
How about `useTranslation(): UseTranslationResult`. If we need additional options/arguments either list separate arguments, or if it's an object: `useTranslation(opts?: UseTranslationOptions)`

Reply:
Looking at your suggestions, for now I went with `UseTranslationResult` but I am not sure about it. Talking about consistent strategy, I do not want to fully decide something right now. We already talked about me wanting to create some code styling rules and I think this could be resolved with that because it needs some investigating and thinking to come up with a good solution (I created an issue for that whole topic #779). What do you think?